clients/fides-js/src/lib/consent.tsx

```diff
@@ -20,6 +19,18 @@ export const initOverlay = async ({
   debugLog(options.debug, "Initializing Fides consent overlays...");
 
   async function renderFidesOverlay(): Promise<void> {
+    // Check if there are any notices within the experience that do not have a user preference
+    const noticesWithNoUserPreferenceExist: boolean = Boolean(
+      experience?.privacy_notices?.some(
+        (notice) => notice.current_preference == null
+      )
+    );
+    const modalLinkId = options.modalLinkId || "fides-modal-link";
+    const modalLinkEl = document.getElementById(modalLinkId);
+    // If we don't have new notices and modal link does not exist, don't render overlay
+    if (!noticesWithNoUserPreferenceExist && !modalLinkEl) {
+      return Promise.resolve();
+    }
```

Comment:
I have a feeling there's a bug here... it's a race condition. What if the link renders to the page after our render method fires?

I think the safest thing to do is not gate _any_ overlay rendering logic on the existence of that link, and throw it into a delayed interval - wait like 200ms and then check the DOM if it exists and add the click handler then, etc.

Sorry!

You can carve that out into a follow-up issue though, as long as it's not going to cause any other problems?

Reply:
We already wait until the DOM is completely loaded before calling our `renderFidesOverlay()` method within this file, so I think we should be safe?

Reply:
I went ahead and create a ticket to track, but we can close out if not needed- https://github.com/ethyca/fides/issues/3483